File: clima python/principalclima.py
from clima import clima

lista = []

linha = "2020, Janeiro, Quente, muita"
dadosLinha = linha.split(",")

obj_clima = clima(dadosLinha[0], dadosLinha[1], dadosLinha[2], dadosLinha[3])

if obj_clima not in lista:
    lista.append(obj_clima)

linha = "2020, Janeiro, Frio, pouca" 
dadosLinha = linha.split(",")

obj_clima = clima(dadosLinha[0], dadosLinha[1], dadosLinha[2], dadosLinha[3])

if obj_clima not in lista:
    lista.append(obj_clima)


for c in lista:
    print(c)

File: clima java/principalclima.java
import java.util.ArrayList;

public class principalclima {
    public static void main(String[] args) {
        ArrayList<clima> lista = new ArrayList<>();

        //simulando que estamos lendo linha a linha do arquivo
        String linha = "2020,Janeiro,Quente,muita";
        String dadosLinha[] = linha.split(",");
        clima objClima = new clima(dadosLinha[0], dadosLinha[1], dadosLinha[2], dadosLinha[3]);
        

        if(!lista.contains(objClima)){
            lista.add(objClima);
        }
        
        linha = "2020,Janeiro,Frio,pouca";
        dadosLinha = linha.split(",");
        objClima = new clima(dadosLinha[0], dadosLinha[1], dadosLinha[2], dadosLinha[3]);

        if(!lista.contains(objClima)){
            lista.add(objClima);
        }

        for (clima c : lista){
            System.out.println(c);
        }


      
    }
}
